source/MaterialXCore/Element.cpp

```diff
@@ -104,6 +104,24 @@ string Element::getNamePath(ConstElementPtr relativeTo) const
     return res;
 }
 
+ElementPtr Element::getElementByNamePath(const std::string& path)
```

Comment:
Since this method can only find descendants of the given element, what would you think of the method name "getDescendant"?  To my mind, the name path is the only input that can uniquely qualify an element's descendant, so perhaps the "byNamePath" extension is unneeded.

Reply:
I had voted for using "descendent" as well :)

Reply:
Haha, ya, we had some discussions about this. I like "getDescendant". It's more precise.